File: app/src/main/java/dev/yiray/qickmvivm/retrofit/UserService.java
package dev.yiray.qickmvivm.retrofit;

import android.app.Application;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class UserService {
    private static final String BASE_URL = "https://api.github.com/";
    private static final Retrofit retrofit = new Retrofit.Builder()
            .baseUrl(BASE_URL)
            .addConverterFactory(GsonConverterFactory.create())
            .build();

    public static UserInterface getInterface() {
        return retrofit.create(UserInterface.class);
    }
}

File: app/src/main/java/dev/yiray/livedatamvi/retrofit/UserService.kt
package dev.yiray.livedatamvi.retrofit

import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

object UserService {
    private const val BASE_URL = "https://api.github.com/"
    private val retrofit = Retrofit.Builder()
        .baseUrl(BASE_URL)
        .addConverterFactory(GsonConverterFactory.create())
        .build()
    val api: UserApi
        get() = retrofit.create(UserApi::class.java)
}